gimie/make_graph.py

```diff
@@ -2,30 +2,54 @@
 import hashlib
 from calamus import fields
 from calamus.schema import JsonLDSchema
+from urllib.parse import urlparse
 
 schema = fields.Namespace("http://schema.org/")
 
 
+def validate_url(url: str):
+    """Checks if input is a valid URL.
+    credits: https://stackoverflow.com/a/38020041
+
+    Examples
+    -------------
+    >>> validate_url('/data/my_repo')
+    False
+    >>> validate_url(532)
+    False
+    >>> validate_url('https://www.github.com/SDSC-ORD/gimie')
+    True
+    >>> validate_url('http://www.github.com/SDSC-ORD/gimie')
+    True
+    >>> validate_url('github.com/SDSC-ORD/gimie')
+    True
+    """
+    # if "https://" or "http://" not in url:
+    #     url = "https://" + url
+    try:
+        result = urlparse(url)
+        return all([result.scheme, result.netloc])
+    except:
+        return False
+
+
 def generate_fair_uri(
     repository_path: str = "https://example.com/Repository/",
```

Comment:
I'm not sure it makes sense to provide a default value here (when would we want use it?).
If the purpose is to show what the input is expected to look like, this would fit better in the Examples section of the docstring (similar to validate_url, with example inputs and outputs.)

Reply:
You're right. This would not make any sense in a real scenario, since the functions would not find "example.com/Repository" and thus the whole thing would do anything. This worked only in my example because I hardcoded some values (the ones I have TODO behind 😅 ). I will remove it.